uwblueprint/feeding-canadian-kids
# Meal Donor Dashboard Create Meal Request Form First Step

## Notion Ticket Link

[Replace with Ticket URL](https://www.notion.so/uwblueprintexecs/Meal-Donation-Form-Step-1-UI-e56c737630f64b6cb70bc944f08924d9?pvs=4)



## Implementation Description
* Implemented a basic three step form structure for the Meal Donor Dashboard
* Created a versatile "ThreeStepFormTitle" component that can be used for all three step forms
* Implemented the "Meal Donation Form" page (first panel of the Meal Donor Dashboard three step form) which retrieves data from the "getMealRequestsByIds" query
* NOTE: in ASPDashboard.tsx, query parameter values are added to the page's URL to retrieve test data from graphql. The values in the query parameter are temporary and only there for testing purposes.



## Steps To Test
1. Go to the Home Page -> Old Dashboard
2. Click on the "Meal Donation Form" button located of the dashboard
3. To test if the page can retrieve data from different meal ids, modify line 62 of the ASPDashboard.tsx file with the ids of your choice. NOTE: you must go back to the home page and repeat steps 1 and 2 again to see the updated data as the page URL changes. 


- Line 62 (for reference):  ``<NavigationButton path={`${Routes.MEAL_DONOR_FORM_PAGE}?ids=65cc28ce55434ad75a3a6439,65cc28cd55434ad75a3a6437,65cc28ce55434ad75a3a6438`}>Meal Donation Form</NavigationButton>``



## What Should Reviewers Focus On?
* Focus on the overall design, spacing, and format of the page and let me know if there's anything I should change :) 

## Checklist
- [ ✅] My PR name is descriptive and in imperative tense
- [✅ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [✅] I have run the appropriate linter(s)
- [✅ ] I have added tests for my changes
- [ ✅] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
